components/formatters/src/number-formatter.ts

```diff
@@ -34,39 +34,42 @@ const ABBREVIATION_LEVELS = [
   { multiplier: KILO_MULTIPLIER, postfix: 'k' },
 ];
 
-const MIN_VALUE = 0.001;
-// tslint:enable:no-magic-numbers
-
 /**
  * Helper function that adjusts a given number by abbreviating or adjusting the precision
  * Returns adjusted number as a string
  *
  * @param value - numeric value to be transformed
  * @param abbreviate - defines whether to abbreviate big numbers or not (by adding appropriate postfixes); false by default
+ * @param maxPrecision - The maximum amount of digits to be used, if provided
  *
  */
 export function adjustNumber(
   value: number,
   abbreviate: boolean = false,
+  maxPrecision?: number,
 ): string {
   const calcValue = Math.abs(value);
   return abbreviate && calcValue >= KILO_MULTIPLIER
     ? abbreviateNumber(value)
-    : adjustPrecision(value);
+    : adjustPrecision(value, maxPrecision);
 }
 
-function adjustPrecision(value: number): string {
+function adjustPrecision(value: number, maxPrecision?: number): string {
   // tslint:disable:no-magic-numbers
   const calcValue = Math.abs(value);
+  const minValue = 1 / Math.pow(10, Math.max(maxPrecision || 3, 0));
```

Comment:
NIT: can we define the `3` as constant so that it is somehow clear that this is the fallback?

Reply:
Thx @lukasholzer, I have udpated the pr accordingly